nucypher/blockchain/eth/clients.py

```diff
@@ -54,69 +55,124 @@ def __new__(self, w3: Web3, *args):
         GANACHE = 'EthereumJS TestRPC'
 
         try:
-            gclass = {
+            cls = {
                 GETH: GethClient,
                 PARITY: ParityClient,
                 GANACHE: GanacheClient,
             }[node_technology]
-            return gclass(w3, *client_data)
-
         except KeyError:
             raise NotImplementedError(node_technology)
 
+        return cls(w3, *client_data)
```

Comment:
This is what I'm worried about failing with the 3 or 4 value split. Currently, I think this line will fail for Geth and Parity.